lib/db_utils.go

```diff
@@ -8644,55 +8692,97 @@ func DBGetUserAssociationIdsByAttributes(handle *badger.DB, snap *Snapshot, asso
 			associationQuery.TargetUserPKID == nil ||
 			associationQuery.AssociationType == "" ||
 			associationQuery.AssociationValue == "" {
-			return nil, errors.New("DBGetUserAssociationIdsByAttributes: invalid query params")
+			return nil, nil, errors.New("DBGetUserAssociationIdsByAttributes: invalid query params")
 		}
 		keyPrefix = append(keyPrefix, associationQuery.AppPKID.ToBytes()...)
 	}
 
+	// Convert LastSeenAssociationID to LastSeenKey.
+	var lastSeenKey []byte
+	if associationQuery.LastSeenAssociationID != nil {
+		lastSeenKey, err = _dbUserAssociationIdToKey(handle, snap, associationQuery.LastSeenAssociationID, prefixType)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "DBGetUserAssociationIdsByAttributes: ")
+		}
+	}
+
+	// Map deleted UTXO AssociationIDs to keys.
+	deletedUtxoKeys := NewSet[string]([]string{})
+	for _, deletedUtxoAssociationId := range deletedUtxoAssociationIDs.ToSlice() {
+		deletedUtxoKey, err := _dbUserAssociationIdToKey(handle, snap, deletedUtxoAssociationId, prefixType)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "DBGetUserAssociationIdsByAttributes: ")
+		}
+		if deletedUtxoKey != nil {
```

Comment:
when would deletedUtxoKey ever be nil?

Reply:
If there is no AssociationEntry found for that AssociationID, then deletedUtxoKey will be nil. This can happen and a unit test caught this edge case when an association is deleted in the UTXO view and is never persisted in the database. In that case, we don't need to include the key in our set of deleted keys, since the association never made it to the database (and it won't be returned from the UTXO view either since it's deleted).